slam_toolbox/lib/karto_sdk/src/Mapper.cpp

```diff
@@ -1605,7 +1605,11 @@ namespace karto
     // only attach link information if the edge is new
     if (isNewEdge == true)
     {
-      pEdge->SetLabel(new LinkInfo(pFromScan->GetSensorPose(), rMean, rCovariance));
+      auto corrected_pose=pToScan->GetCorrectedPose();
+      pToScan->SetSensorPose(rMean);
+      auto corrected_mean_pose=pToScan->GetCorrectedPose();
+      pToScan->SetCorrectedPose(corrected_pose);
```

Comment:
L1608 is `corrected_pose=pToScan->GetCorrectedPose();` but then you `pToScan->SetCorrectedPose(corrected_pose);`, how is L1611 meaningful? It looks like you just `set` what you `get` without changing.

Reply:
L1609 changes m_CorrectedPose of pToScan. I'm not sure whether the original m_CorrectedPose value is going to be used in processes below, so I simply reset  the m_CorrectedPose of pToScan to make sure every value is the same as the original code before calling pEdge->SetLabel.

Reply:
Can you check and see?